### PR TITLE
Checkout submodules for HyTeg

### PR DESCRIPTION
**Pull request by vulder**

### CODE REVIEW

File: varats/varats/projects/cpp_projects/hyteg.py [hunk 1]
```diff
@@ -19,6 +19,7 @@
 from varats.project.sources import FeatureSource
 from varats.project.varats_command import VCommand
 from varats.project.varats_project import VProject
+from varats.utils.git_commands import init_all_submodules, update_all_submodules
```
Comment (vulder) [inline varats/varats/projects/cpp_projects/hyteg.py]:
```suggestion
from varats.utils.git_commands import update_all_submodules
```

File: varats/varats/projects/cpp_projects/hyteg.py [hunk 2]
```diff
@@ -86,25 +110,37 @@ def compile(self) -> None:
 
         mkdir("-p", hyteg_source / "build")
 
+        update_all_submodules(hyteg_source, recursive=True, init=True)
```
Comment (LuAbelt) [inline varats/varats/projects/cpp_projects/hyteg.py]: From what I see, HyTeg is also doing this via CMake, but it cannot hurt to also have it here

**Review by LuAbelt — APPROVED** [review]

File: varats/varats/projects/cpp_projects/hyteg.py [hunk 3]
```diff
@@ -86,25 +110,37 @@ def compile(self) -> None:
...
+
```
Comment (vulder) [inline varats/varats/projects/cpp_projects/hyteg.py]:
```suggestion
```

File: varats/varats/projects/cpp_projects/hyteg.py [hunk 4]
```diff
@@ -19,24 +21,46 @@
 from varats.project.sources import FeatureSource
 from varats.project.varats_command import VCommand
 from varats.project.varats_project import VProject
+from varats.utils.git_commands import init_all_submodules, update_all_submodules
```
Comment (vulder) [inline varats/varats/projects/cpp_projects/hyteg.py]:
```suggestion
```

**Pull request merged**